nautobot_golden_config/utilities/helper.py

```diff
@@ -39,26 +51,28 @@ def get_job_filter(data=None):
     elif data.get("device"):
         query.update({"id": data["device"].values_list("pk", flat=True)})
 
-    base_qs = Device.objects.none()
-    for obj in models.GoldenConfigSetting.objects.all():
-        base_qs = base_qs | obj.get_queryset().distinct()
+    init_qs = DeviceFilterSet(data=query, queryset=Device.objects.all()).qs
 
-    if base_qs.count() == 0:
+    if init_qs.count() == 0:
         raise NornirNautobotException(
             "The base queryset didn't find any devices. Please check the Golden Config Setting scope."
         )
-    devices_filtered = DeviceFilterSet(data=query, queryset=base_qs)
-    if devices_filtered.qs.count() == 0:
+
+    settings_map = get_device_to_settings_map(init_qs)
+    dev_ids = [str(i) for i in set(settings_map.keys())]
```

Comment:
Hmm, how big could this get?

Reply:
Not realistic, just wanting to see if we can get eyes on it.